continuous_integration/flake8_diff.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+set -e
+
+# Travis does the git clone with a limited depth (50 at the time of
+# writing). This may not be enough to find the common ancestor with
+# $REMOTE/master so we unshallow the git checkout
+git fetch --unshallow || echo "Unshallowing the git checkout failed"
```

Comment:
I haven't tried yet. The thing to bear in mind is that the Travis build that does the flake8 checks doesn't run tests so I would expect the unshallowing not to matter that much in practice. Also note that if you run this script locally the unshallowing won't need to happen (at least in most cases).